parquet/src/main/java/org/apache/iceberg/parquet/ReadConf.java

```diff
@@ -166,17 +167,22 @@ ParquetFileReader reader() {
   }
 
   private Map<Long, Long> generateOffsetToStartPos() {
-    ParquetFileReader fileReader = newReader(this.file, ParquetReadOptions.builder().build());
-    Map<Long, Long> offsetToStartPos = new HashMap<>();
-    long curRowCount = 0;
-    for (int i = 0; i < fileReader.getRowGroups().size(); i += 1) {
-      BlockMetaData meta = fileReader.getRowGroups().get(i);
-      offsetToStartPos.put(meta.getStartingPos(), curRowCount);
-      curRowCount += meta.getRowCount();
-    }
+      try (ParquetFileReader fileReader = newReader(file, ParquetReadOptions.builder().build())) {
+        Map<Long, Long> offsetToStartPos = new HashMap<>();
```

Comment:
Maps.newHashMap() was not auto resolved by imports so going to use default constructor